### PR TITLE
refactor: usages of static seo to layout props

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Basically migration of pages to our new flow. With this system, we can be confident that the SEO information will be rendered regardless of how early we return `null` on certain pages based on some conditions.
- One thing to look out for to make it easier to review: on pages where we moved the SEO, you must see equivalent deleted lines of code from the same page.

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://migrate-seo-usages.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/graphql/squads.ts [hunk 1]
```diff
@@ -243,6 +243,19 @@ export const SQUAD_STATIC_FIELDS_QUERY = gql`
...
+  'id' | 'name' | 'public' | 'description' | 'image'
+>;
+
+export const getSquadStaticData = async (
+  handle: string,
+): Promise<SquadStaticData> => {
+  const res = await gqlClient.request(SQUAD_STATIC_FIELDS_QUERY, { handle });
+
+  return res.source;
+};
```
Comment (sshanzel) [inline packages/shared/src/graphql/squads.ts:257]: Just moved it as a generic function, I believe this is used somewhere else, too, but I didn't want to change it here as this is already a big PR.

File: packages/shared/src/graphql/users.ts [hunk 2]
```diff
@@ -398,6 +398,18 @@ export const GET_REFERRING_USER_QUERY = gql`
...
+): Promise<UserShortProfile> => {
+  try {
+    const res = await gqlClient.request(GET_REFERRING_USER_QUERY, { id });
+
+    return res.user;
+  } catch (err) {
+    return undefined;
+  }
+};
+
```
Comment (sshanzel) [inline packages/shared/src/graphql/users.ts:412]: Same here.

**Comment by sshanzel** [comment]
Previews:

Screenshot 2024-10-25 at 6 49 49 PM

Notice how it wasn't working on the target branch. Because we applied the fix (migration) here.
Screenshot 2024-10-25 at 6 50 14 PM

**Comment by sshanzel** [comment]
For user profile page:

Screenshot 2024-10-25 at 6 52 08 PM

Feel free to review 🥳 cc @dailydotdev/web-team

**Review by ilasw — APPROVED** [review]
Seems good! Nice!

**Pull request merged**